plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemoryMetadata.java

```diff
@@ -153,6 +153,37 @@ public synchronized void dropSchema(ConnectorSession session, String schemaName,
         verify(schemas.remove(schemaName));
     }
 
+    @Override
+    public synchronized void renameSchema(ConnectorSession session, String source, String target)
+    {
+        schemas.remove(source);
+        schemas.add(target);
+
+        for (Map.Entry<SchemaTableName, Long> table : tableIds.entrySet()) {
+            if (table.getKey().getSchemaName().equals(source)) {
+                tableIds.put(new SchemaTableName(target, table.getKey().getTableName()), table.getValue());
```

Comment:
Why we don't have to remove entries, just adding new ones? Also, if this is a map, why iterate at all?

Reply:
The key contains both schema and table names. How do we get tables under a specific schema without iteration?

Reply:
Ah, this copies multiple tables into the new schema. What about removing them from the old schema?

Reply:
They're removed in the latest change: https://github.com/trinodb/trino/compare/d4a6828b30c1928d456fe65d850fd7d78f396521..9bedd37ab98c5e9fbafc3eb74d0099e3b7494d3f